<launch>

  <arg name="tag_id"                            default="5"/>
  <arg name="tag_size"                          default="0.02"/>
  <arg name="tag_family"                        default="tagStandard41h12"/>
  <arg name="standalone_tags"                   default="[{id: $(arg tag_id), size: $(arg tag_size), name: ar_tag}]"/>
  <arg name="camera_frame"                      default="camera_color_optical_frame"/>
  <arg name="apriltag_ns"                       default="apriltag"/>
  <arg name="camera_color_topic"                default="camera/color/image_raw"/>
  <arg name="camera_info_topic"                 default="camera/color/camera_info"/>
  <arg name="image_save_dir"                    default="interbotix/picture_snapper/"/>

  <param name="$(arg apriltag_ns)/camera_color_topic" value="$(arg camera_color_topic)"/>
  <param name="$(arg apriltag_ns)/camera_info_topic" value="$(arg camera_info_topic)"/>

  <node
    pkg="apriltag_ros"
    type="apriltag_ros_single_image_server_node"
    name="ar_tracker"
    clear_params="true"
    output="screen"
    ns="$(arg apriltag_ns)">
    <param name="tag_family"                      value="$(arg tag_family)"/>
    <param name="tag_threads"                     value="2"/>
    <param name="tag_decimate"                    value="1.0"/>
    <param name="tag_blur"                        value="0.0"/>
    <param name="tag_refine_edges"                value="1"/>
    <param name="tag_debug"                       value="0"/>
    <param name="publish_tf"                      value="false"/>
    <param name="remove_duplicates"               value="true"/>
    <param name="camera_frame"                    value="$(arg camera_frame)"/>
    <rosparam param="standalone_tags"             subst_value="True">$(arg standalone_tags)</rosparam>
    <rosparam param="tag_bundles">[]</rosparam>
  </node>

  <node
    name="picture_snapper"
    pkg="interbotix_perception_modules"
    type="picture_snapper"
    output="screen"
    ns="$(arg apriltag_ns)">
    <param name="image_save_dir"                  value="$(arg image_save_dir)"/>
  </node>

</launch>
